darwin/torch/utils.py

```diff
@@ -4,12 +4,44 @@
 from typing import Iterable, List, Optional, Tuple
 
 import numpy as np
+import torch
+from upolygon import draw_polygon
+
 from darwin.cli_functions import _error, _load_client
 from darwin.dataset.identifier import DatasetIdentifier
 from darwin.datatypes import Segment
-from upolygon import draw_polygon
 
-import torch
+
+def flatten_masks_by_category(
+    masks: torch.Tensor, cats: torch.Tensor, remove_overlap: bool = False, overlap: int = 255
+) -> torch.Tensor:
+    """
+    Takes a list of masks and flattens into a single mask output with category id's overlaid into one tensor.
+    By default overlapping sections of masks are replaced with the numerically largest category id.
+    Parameters
+    ----------
+    masks : torch.Tensor
+        lists of masks with shape [x, image_height, image_width] where x is the number of categories
+    cats : torch.Tensor
+        tensor vector of category id's with shape [x]
+    remove_overlap : bool, optional
+        overlapping areas will be replaced with default_overlap value, by default False
+    overlap : int, optional
+        value to replace the overlapping segments with, used in conjuction with remove_overlap, by default 255
+
+    Returns
+    -------
+    torch.Tensor
+        Flattened mask of category id's
+    """
+    if masks.shape[0] != cats.shape[0]:
```

Comment:
Feels maybe more like a `ValueError`?  If it were and `AssertionError`, you'd expect it to come from an `assert`, and you could get that with just
```assert masks.shape[0] != cats.shape[0]```

Also feels a bit like it could expand slightly to be be more fault-tolerant, given that this is SDK use code, to test that these are `torch.Tensor` on the way in (type hinting will redline that, but the function signature won't prevent you calling it with a different object at runtime)

Reply:
My thinking here is that yeah I would have normally used an assertion, but can send in a custom message in case this ever hits a customer call. Re torch types, this function is mostly internal to the classes that we know generate torch types. I'm unclear if it would find use in the wider SDK but I'll add the checking in case it does.